### PR TITLE
Add X-Application-ID header

### PR DESCRIPTION
**Pull request by brianesquilona**
X-Application-ID header is needed when using JWT.
 In fact api-key will not be authorize to use the tunnel, should use a formed jwt.

### CODE REVIEW

File: main.go
Comment (sk-arm) [inline main.go:23]: Is application id a required parameter? Should we check whether it is empty?

Comment (brianesquilona) [reply main.go:23]: This might go away if the application id is embedded in the jwt anyway, but more of architecture discussion. @zhucebuliaolongchuan what do you think?

Comment (zhucebuliaolongchuan) [reply main.go:23]: Yeah this won't be needed if application-id could be passed through already. But I am thinking about that if we could let the customers to use this client to sign their claims as well so they don't need to do that themselves. 

So if we want to chase that way, we need to have two options for them. First option is they could provide the key directly(they sign the claims themselves), or they could provide their private key only and then we could generate the jwt within this client.

Any ideas? @sk-arm @brianesquilona

Comment (sk-arm) [reply main.go:23]: Hi Brian, until they make that architecture change we should go with this approach. We should track that architecture change separately and adjust the code accordingly when it happens. Otherwise we will be in waiting mode. 

Hi Chuan, is there an item in the JIRA to track that architecture change?

Chuan, regarding signing the JWT using their private key,  I think the key idea is customer never need to share their private key (they may have a secure store for their private key) with us (our tools, even if it is in their local machine) and that option may not acceptable. We can certainly ask PM.

Comment (zhucebuliaolongchuan) [reply main.go:23]: The architecture change is already in place, while it doesn't affect the client side really since `x-application-id` is a header being passed after the token is verified and it only happens between IAM service and our cloud service(fog-proxy-cloud), so I don't think we should worry about this here.

Yeah. This sounds more like a Product decision.

Comment (sk-arm) [reply main.go:23]: @zhucebuliaolongchuan what is the conclusion? If we we plan to keep the application-id as command line we should go ahead with this change.

Comment (zhucebuliaolongchuan) [reply main.go:23]: @sk-arm It seems like we need to provide this. Maybe we could name it `application-id`? As the user, they don't need to know the name of the header. IMO `application-id` is more user friendly.

**Comment by zhucebuliaolongchuan** [comment]
@brianesquilona @sk-arm As what we have discussed before, this repo is public. We can't merge the change to the master until all of our changes are being merged to production.

File: docs/Access Control with JWT Token.md
Comment (zhucebuliaolongchuan) [inline docs/Access Control with JWT Token.md:5]: Because the original API token is also acceptable, I think we should explicitly say that we are supporting two different flows.

File: docs/Access Control with JWT Token.md
Comment (zhucebuliaolongchuan) [inline docs/Access Control with JWT Token.md:31]: This is the detail about how pelion-direct-connect-client deals with the API. I don't think we need to mention this here.

File: docs/Access Control with JWT Token.md
Comment (zhucebuliaolongchuan) [inline docs/Access Control with JWT Token.md:38]: We should put some notes here to specify if this field is provided, the service will check the equivalence between the value of this and the request path parameters. If they are not matched,  the request will be rejected.

**Review by zhucebuliaolongchuan — CHANGES_REQUESTED** [review]

File: main.go
Comment (brianesquilona) [reply main.go:68]: Authentication endpoint says ok so should be a problem.

File: docs/Access Control with JWT Token.md
Comment (brianesquilona) [reply docs/Access Control with JWT Token.md:5]: Yes, but this document is only for jwt token.

Comment (zhucebuliaolongchuan) [reply docs/Access Control with JWT Token.md:5]: Is this the only doc that we are giving out?

Comment (brianesquilona) [reply docs/Access Control with JWT Token.md:5]: I think we should delete this doc once the official document of the tunnel with is out, then put that documentation in the readme. This does not serve as a pelion-direct-connect-client doc.

Comment (zhucebuliaolongchuan) [reply docs/Access Control with JWT Token.md:5]: sounds good. Then never mind. Current info should be good enough

**Review by zhucebuliaolongchuan — APPROVED** [review]
Good work man. LGTM.

**Review by sk-arm — APPROVED** [review]
Looks good.

**Pull request merged**
